src/tree.rs

```diff
@@ -660,6 +661,32 @@ impl<T> Node<T> {
     }
 }
 
+impl<T> Node<T> {
+    /// Iterates over the tree and calls the given visitor function
+    /// with fully resolved path and its value.
+    pub fn for_each<V: FnMut(String, T) -> bool >(self, mut visitor: V)  {
+        let mut queue = VecDeque::new();
+        queue.push_back(self);
+        while let Some(mut node) = queue.pop_front() {
+            if node.children.is_empty() {
+                denormalize_params(&mut node.prefix, &node.remapping);
+                let path = String::from_utf8(node.prefix.into_unescaped()).unwrap();
+                let value = node.value.take().map(UnsafeCell::into_inner);
+                if !visitor(path, value.unwrap()) {
+                    return;
+                }
+            } else {
```

Comment:
I don't think the `else` condition here is correct. It is possible for a node with children to have a value, if the routes `/foo` and `/foo/bar` were added for example. The visitor should be called if the node has a value, independent of the node's children. Might be helpful to add a test with nested routes for this case.